data_collection/gazette/spiders/base/portalgov.py

```diff
@@ -0,0 +1,75 @@
+import re
+from datetime import datetime as dt
+
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class BasePortalGovSpider(BaseGazetteSpider):
+    power = "executive"
+
+    def start_requests(self):
+        domain = self.allowed_domains[0]
+        yield scrapy.FormRequest(
+            url=f"https://{domain}/controllers/diario_oficial/class_diario.php",
+            method="POST",
+            formdata={
+                "func": "5",
+                "param": "1",
+            },
+        )
+
+    def parse(self, response):
+        domain = self.allowed_domains[0]
+        BASE_URL = f"https://{domain}/arquivos/diario_oficial/"
+
+        for gazette_data in response.json():
+            raw_gazette_date = gazette_data["data"]
+            gazette_date = dt.strptime(raw_gazette_date, "%d/%m/%Y").date()
+            if gazette_date > self.end_date:
+                continue
+            if gazette_date < self.start_date:
+                return
+
+            gazette_item = self.create_gazette_item(
+                BASE_URL, gazette_data, gazette_date
+            )
+
+            yield scrapy.Request(
+                gazette_item["file_urls"][0],
+                method="GET",
+                callback=self.parse_valid_gazette_file,
+                cb_kwargs={"gazette": gazette_item},
+            )
+
+    def parse_valid_gazette_file(self, response, gazette):
+        content_length = response.headers.get("Content-Length")
+        if content_length and int(content_length) > 0:
+            yield gazette
```

Comment:
Não precisa implementar esse método. A não ser que estejamos subindo arquivos "vazios" (e seus metadados) no banco. Mas nesse caso (observei 2018-05-14) o arquivo não é baixado e ocorre um erro de validação, o que seria o esperado.

Reply:
Exatamente. Esse erro seria o equivalente a erros de "404 - Not found". A ideia é deixar acontecer mesmo e o log reportar que aquilo tá acontecendo. Eventualmente, isso pode ser usado para pedir que o problema seja corrigido na fonte.

Reply:
Obrigado, @ogecece! 

Removi o método.